.github/workflows/build-and-benchmark-x86.yml

```diff
@@ -0,0 +1,47 @@
+name: build and benchmark on x86-64
+on: [push, pull_request]
+jobs:
+    benchmark-on-self-hosted-x86-64:
+        runs-on: [self-hosted, linux, X64]
+        steps:
+            - name: git checkout
+              uses: actions/checkout@v4
+            - name: release build this PR
+              run:  nice cargo build --release
```

Comment:
Why do we need `nice` for compiling?

Reply:
Because this machine has interactive users which we want to bother as little as possible.